### PR TITLE
Resimplify Makefile

### PR DESCRIPTION
**Pull request by miska**

### CODE REVIEW

**Comment by jimklimov** [comment]
I do not quite agree with loss of variables for program names - we did some renaming and it had to be repeated several times. Also I hope to complete suffix rules or callable functions and so avoid some repetitive code.

**Comment by hintjens** [comment]
Yes, I think it's time to make a library with some reusable code.

File: Makefile [hunk 1]
```diff
@@ -1,23 +1,8 @@
...
-
-PROGRAMS_C = $(PROGRAM_EMAIL) $(PROGRAM_UPS) $(PROGRAM_ZYRE_TRIVIAL)
-# $(PROGRAM_MALAMUTEZ)
-PROGRAMS_CXX = $(PROGRAM_MON) $(PROGRAM_UPSXX)
-PROGRAMS = $(PROGRAMS_C) $(PROGRAMS_CXX)
-
-#SOURCES_C = $(addsuffix .c,$(PROGRAMS_C))
-#SOURCES_CXX = $(addsuffix .cc,$(PROGRAMS_CXX))
-
-RM = rm -f
```
Comment (jimklimov) [inline Makefile:16]: Apparently lack or disablement of this built-in definition is what broke for those who could not use "make clean", weird...

File: Makefile [hunk 2]
```diff
@@ -53,12 +38,3 @@ check-zmq-hpp-presence:
 	    exit $$RES; \
 	fi
 
-testz: $(PROGRAM_ZYRE_TRIVIAL)
```
Comment (jimklimov) [inline Makefile:56]: Fine with this :)

**Comment by vyskocilm** [comment]
it conflicts, please fix or drop

**Pull request closed without merging**